Sources/Extensions/SwiftStdlib/DictionaryExtensions.swift

```diff
@@ -110,6 +110,25 @@ public extension Dictionary {
 
 }
 
+// MARK: - Methods (Value: Equatable)
+public extension Dictionary where Value: Equatable {
+    
+    /// SwifterSwift: Returns an array of all keys that have the given value in dictionary (if applicable).
+    ///
+    ///        let dict = ["key1": "value1", "key2": "value1", "key3": "value2"]
+    ///        dict.keys(forValue: "value1") -> Optional(["key1", "key2"])
+    ///        dict.keys(forValue: "value2") -> Optional(["key3"])
+    ///        dict.keys(forValue: "value3") -> nil //no key that has the value "value3"
+    ///
+    /// - Parameter forValue: Value for which keys are to be fetched.
+    /// - Returns: An array containing keys that have the given value (if applicable).
+    public func keys(forValue value: Value) -> [Key]? {
+        let result = self.compactMap{ $1 == value ? $0 : nil }
```

Comment:
I just did some basic testing and it turns out that the code is sped up significantly by instead using:
```
func keys(forValue value: Value) -> [Key] {
    return keys.filter { self[$0] == value }
}
```
It turns out to be almost twice as fast. I'm guessing that there's some extra-heavy lifting going on in `compactMap`, because I would also have assumed that that would be faster.

Reply:
Oh cool, thanks a lot! I'll make the change